registry/storage/driver/inmemory/mfs.go

```diff
@@ -97,8 +98,13 @@ func (d *dir) list(p string) ([]string, error) {
 		return nil, errIsNotDir
 	}
 
-	var children []string
-	for _, child := range n.(*dir).children {
+	dir, ok := n.(*dir)
```

Comment:
Why a second check for being a directory with a different form?

Reply:
Ah, I didn't realize we have this type check here https://github.com/distribution/distribution/blob/59fd8656ac3c4f34de9dbb93e9fae90f7f1ec9f2/registry/storage/driver/inmemory/mfs.go#L36.

I will remove and add a comment there instead just in case someone else overlooks like like me

Reply:
Addressed: https://github.com/distribution/distribution/pull/4037/commits/a9d31ec7b911a4247cba6b9acc645b11171bbb07